OpenWeather/App/Home/HomeViewModel.swift

```diff
@@ -64,3 +84,89 @@ final class HomeViewModel: ObservableObject, ErrorHandling {
         }
     }
 }
+
+protocol HomeCapable: CurrentWeatherProviding, LocationNameProviding { }
+
+extension HomeCapable where Self == HomeCapabilities {
+    static var mock: Self {
+        HomeCapabilities(
+            locationProviding: MockLocationProvider(),
+            weatherProviding: MockWeatherProvider()
+        )
+    }
+}
+
+struct HomeCapabilities: HomeCapable {
+    private var locationProviding: LocationProviding
+    private var weatherProviding: WeatherProviding
+
+    init(locationProviding: LocationProviding, weatherProviding: WeatherProviding) {
+        self.locationProviding = locationProviding
+        self.weatherProviding = weatherProviding
+    }
+
+    func locationName(for location: CLLocation) async throws -> String {
+        try await locationProviding.locationName(for: location)
+    }
+
+    func currentWeather(for location: CLLocation) async throws -> DeviceWeather {
+        try await weatherProviding.currentWeather(for: location)
+    }
+}
```

Comment:
Ah yes this could just live inside of the HomeViewModel! Then we don't need the protocol and extension.